hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/DispatchingEvmFrameState.java

```diff
@@ -518,23 +542,29 @@ public Optional<ExceptionalHaltReason> tryTrackingSelfDestructBeneficiary(
         }
         final AccountID accountID = AccountID.newBuilder().accountNum(number).build();
         final var account = nativeOperations.getAccount(accountID);
-        if (account == null) {
-            final var token = nativeOperations.getToken(number);
-            if (token != null) {
-                // If the token is deleted or expired, the system contract executed by the redirect
-                // bytecode will fail with a more meaningful error message, so don't check that here
-                return new TokenEvmAccount(address, this);
+        if (account != null) {
+            if (account.deleted() || account.expiredAndPendingRemoval() || isNotPriority(address, account)) {
```

Comment:
I believe if the address given is an evm address and a "normal" (long zero) address exists for the account, we want the user to provide the normal address rather than the evm address.

Reply:
oh so this is one place we enforce that, TY